import_obfuscation.go

```diff
@@ -62,6 +67,62 @@ func appendPrivateNameMap(pkg *goobj2.Package, nameMap map[string]string) error
 	return nil
 }
 
+func extractDebugObfSrc(pkgPath string, pkg *goobj2.Package) error {
+	if envGarbleDebugDir == "" {
+		return nil
+	}
+	for _, member := range pkg.ArchiveMembers {
+		if member.ArchiveHeader.Name != garbleSrcHeaderName {
+			continue
+		}
+
+		osPkgPath := filepath.FromSlash(pkgPath)
+		pkgDebugDir := filepath.Join(envGarbleDebugDir, osPkgPath)
+		if err := os.MkdirAll(pkgDebugDir, 0o755); err != nil {
+			return err
+		}
+
+		archiveSize, err := strconv.Atoi(member.ArchiveHeader.Date)
+		if err != nil {
+			return err
+		}
+
+		archiveBytes := member.ArchiveHeader.Data[:archiveSize]
+
+		archive := bytes.NewBuffer(archiveBytes)
+		gzipReader, err := gzip.NewReader(archive)
+		if err != nil {
+			return err
+		}
+		defer gzipReader.Close()
+		tarReader := tar.NewReader(gzipReader)
+
+		for {
+			header, err := tarReader.Next()
+			if err == io.EOF {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+
+			debugFilePath := filepath.Join(pkgDebugDir, header.Name)
+			debugFile, err := os.Create(debugFilePath)
+			if err != nil {
+				return err
+			}
+			if _, err := io.Copy(debugFile, tarReader); err != nil {
+				return err
+			}
+			debugFile.Close()
+
+			obfuscationTime := header.ModTime.Local()
+			os.Chtimes(debugFilePath, obfuscationTime, obfuscationTime) // Restore obfuscation time
```

Comment:
surely you should check for the error here?

Reply:
also, move the comment to its own line and expand a bit - why do we want to set the creation and lst modification times? Presumably for other programs or editors?

Reply:
I set the file time so as not to mislead the user about when the file was actually obfuscated. It's an optional feature, but I found it quite convenient.

Reply:
Sure. Please just document the code.